tap_github/streams.py

```diff
@@ -412,3 +401,61 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class IssueEventsStream(GitHubStream):
+    """
+    Defines 'IssueEvents' stream.
+    Issue events are fetched from the repository level (as opposed to per issue)
+    to optimize for API quota usage.
+    """
+
+    name = "issue_events"
+    path = "/repos/{org}/{repo}/issues/events"
+    primary_keys = ["id"]
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        row["issue_number"] = int(row["issue"]["number"])
+        row["issue_url"] = row["issue"]["url"]
```

Comment:
I don't know too much about the ELT process, but Laurent hinted that the output table should be a pure dump of the underlying schema. This looks like just cleanup to make the tap easier to use but just wondering what line we can draw for what is considered extracting vs transforming. This looks good for now tho!